do-not-commit/avoid-insecure-deserialization.py

```diff
@@ -0,0 +1,94 @@
+from django.http import HttpResponse
+import datetime
+
+
+def current_datetime(request):
+    # ok
+    user_obj = request.cookies.get("uuid")
+    now = datetime.datetime.now()
+    html = "<html><body>It is now %s.</body></html>" % now
+
+    return "Hey there! {}!".format(pickle.loads(b64decode(html)))
+
+
+# pickle tests
+
+
+def current_datetime(request):
+    # ruleid:avoid-insecure-deserialization
+    user_obj = b64decode(request.cookies.get("uuid"))
+    now = datetime.datetime.now()
+    html = "<html><body>It is now %s.</body></html>" % now
+
+    return "Hey there! {}!".format(pickle.loads(user_obj))
+
+
+def current_datetime(request):
+    # ruleid:avoid-insecure-deserialization
+    user_obj = request.cookies.get("uuid")
+    now = datetime.datetime.now()
+    html = "<html><body>It is now %s.</body></html>" % now
+
+    return "Hey there! {}!".format(pickle.loads(user_obj))
+
+
+def current_datetime(request):
+    # ruleid:avoid-insecure-deserialization
+    user_obj = request.cookies.get("uuid")
+    now = datetime.datetime.now()
+    html = "<html><body>It is now %s.</body></html>" % now
+
+    return "Hey there! {}!".format(pickle.loads(b64decode(user_obj)))
+
+
+def current_datetime(request):
+    # ruleid:avoid-insecure-deserialization
+    return "Hey there! {}!".format(pickle.loads(b64decode(request.cookies.get("uuid"))))
+
+
+# Other libraries
+
+
+def current_datetime(request):
+    # ruleid:avoid-insecure-deserialization
+    user_obj = b64decode(request.cookies.get("uuid"))
+    now = datetime.datetime.now()
+    html = "<html><body>It is now %s.</body></html>" % now
+
+    return "Hey there! {}!".format(_pickle.loads(user_obj))
```

Comment:
Avoid using `pickle`, which is known to lead to code execution vulnerabilities.
When unpickling, the serialized data could be manipulated to run arbitrary code.
Instead, consider serializing the relevant data as JSON or a similar text-based
serialization format. ([serialization.pickle.avoid-pickle](https://semgrep.dev/r/serialization.pickle.avoid-pickle))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=serialization.pickle.avoid-pickle) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>